src/ResourceManager/Sql/Commands.Sql/Server/Cmdlet/GetAzureSqlServer.cs

```diff
@@ -40,21 +53,39 @@ public class GetAzureSqlServer : AzureSqlServerCmdletBase, IModuleAssemblyInitia
         public string ServerName { get; set; }
 
         /// <summary>
+        /// Intializes the model adapter
+        /// </summary>
+        /// <param name="subscription">The subscription the cmdlets are operation under</param>
+        /// <returns>The server adapter</returns>
+        protected override AzureSqlServerAdapter InitModelAdapter(IAzureSubscription subscription)
+        {
+            return new AzureSqlServerAdapter(DefaultContext);
+        }
+
+        /// <summary>
         /// Gets a server from the service.
         /// </summary>
         /// <returns>A single server</returns>
         protected override IEnumerable<AzureSqlServerModel> GetEntity()
         {
             ICollection<AzureSqlServerModel> results = null;
 
-            if (this.MyInvocation.BoundParameters.ContainsKey("ServerName"))
+            if (MyInvocation.BoundParameters.ContainsKey("ServerName") && MyInvocation.BoundParameters.ContainsKey("ResourceGroupName"))
             {
                 results = new List<AzureSqlServerModel>();
                 results.Add(ModelAdapter.GetServer(this.ResourceGroupName, this.ServerName));
             }
+            else if (MyInvocation.BoundParameters.ContainsKey("ResourceGroupName"))
+            {
+                results = ModelAdapter.ListServersByResourceGroup(this.ResourceGroupName);
+            }
+            else if (!MyInvocation.BoundParameters.ContainsKey("ServerName"))
+            {
+                results = ModelAdapter.ListServers();
+            }
             else
             {
-                results = ModelAdapter.GetServers(this.ResourceGroupName);
+                throw new PSArgumentException("When specifying the serverName parameter the ResourceGroup parameter must also be used");
```

Comment:
Can you split this into 2 ParamGroups so that it's enforced by powershell logic instead of custom code?
Group 1: `Get-AzureRmSqlServer`
Group 2: `Get-AzureRmSqlServer -ResourceGroupName [- ServerName]`